src/connectors/impls/clickhouse.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::connectors::prelude::*;
+
+use clickhouse_rs::{Block, Pool};
+
+#[derive(Default, Debug)]
+pub(crate) struct Builder {}
+
+#[async_trait::async_trait]
+impl ConnectorBuilder for Builder {
+    fn connector_type(&self) -> ConnectorType {
+        "clickhouse".into()
+    }
+
+    async fn build(&self, alias: &str, raw_config: &ConnectorConfig) -> Result<Box<dyn Connector>> {
+        // So something bad happens here and nobody know what or why.
+        let config = raw_config
+            .config
+            .as_ref()
+            .ok_or_else(|| ErrorKind::MissingConfiguration(alias.to_string()))?;
+
+        let config = ClickhouseConfig::new(config)?;
+
+        Ok(Box::new(Clickhouse { config }))
+    }
+}
+
+pub struct Clickhouse {
+    config: ClickhouseConfig,
+}
+
+#[async_trait::async_trait]
+impl Connector for Clickhouse {
+    async fn create_sink(
+        &mut self,
+        sink_context: SinkContext,
+        builder: SinkManagerBuilder,
+    ) -> Result<Option<SinkAddr>> {
+        let db_url = self.connection_url();
+        let sink = ClickhouseSink { db_url, pool: None };
+        builder.spawn(sink, sink_context).map(Some)
+    }
+
+    fn codec_requirements(&self) -> CodecReq {
+        CodecReq::Structured
+    }
+}
+
+impl Clickhouse {
+    fn connection_url(&self) -> String {
+        let ClickhouseConfig {
+            host,
+            port,
+            database,
+            ..
+        } = &self.config;
+        let port = port.unwrap_or(ClickhouseConfig::DEFAULT_PORT);
+
+        format!("tcp://{host}:{port}/{database}?compression=lz4")
+    }
+}
+
+#[derive(Deserialize)]
+struct ClickhouseConfig {
+    host: String,
```

Comment:
We've standardized to using a `url` for all connectors it would be good to do that here too - it's in `connectors::utils::url::Url`

Reply:
This means I have to add a new `TcpDefaults`, similar to [`HttpDefaults`](https://github.com/tremor-rs/tremor-runtime/blob/28d37fcf8adc71cefbc34c18bc204d7aaa27721c/src/connectors/utils.rs#L72), right?

Reply:
The `*Defaults` is used to allow omititng the protocol so that the system can auto-fill them. That's why we have a wrapper around `Url`. For `clickhouse` it seems `tcp` with a default port of `9000` would be a good default? that way people can just write `url = "localhost"` and it's equivalent to `url = "tcp://localhost:9000"` (I think I saw 9k as a default value in the code)

Reply:
@Licenser 

This *should* be fixed by [23b05f8](https://github.com/tremor-rs/tremor-runtime/pull/1538/commits/23b05f8980efa8abeda7c5db104f75fb822a4871), but i'm definitely not sure i did it the correct way; can you make sure I used the `Url` type the way you expected?